books/english-composition/bake

```diff
@@ -54,11 +54,8 @@ recipe = Kitchen::BookRecipe.new(book_short_name: :english_composition) do |doc|
       page.search('div[data-type="media"]').each do |media_container|
         BakeIframes.v1(outer_element: media_container)
       end
-      page.search('section').exercises.each do |exercise|
-        BakeNumberedExercise.v1(exercise: exercise,
-                                number: exercise.count_in(:chapter),
-                                suppress_solution_if: false,
-                                note_suppressed_solutions: false)
+      page.search('section').injected_questions.each do |question|
+        BakeInjectedExerciseQuestion.v1(question: question, number: question.count_in(:chapter))
```

Comment:
@kswanson33 from where did you get a content? I fetched it from `osbooks-writing-guide` repo and there exercises don't have any extra class or data-type. Just `data-type="exercise"`. How can I check locally if it works? 
Also as I see in the kitchen `injected_questions` are exercises with `[data-type='exercise-question']` and in the tests you put exercises with `data-type="injected-exercise"`. Shouldn't selector be overwritten here by this code?

```
doc.selectors.override(
   injected_question: "div[data-type="injected-exercise"]"
  )
```
I'm not sure if I understand correctly what is going on here :smile: 

Oh, and one another thing, section exercises should be reset by section now (https://github.com/openstax/cnx-recipes/issues/4032)

Reply:
Good question -- this is a very new change! It just became part of the pipeline with the last release. Running `./install-submodules.sh --upgrade` in book-pipeline should change the selector to `injected-exercise` and `exercise-question`.

For more background, this comes from a change to the [exercise template](https://github.com/openstax/cnx-epub/blob/master/cnxepub/templates/exercise_template.py) in cnx-epub that converts injected exercise JSON to HTML. The reason why this change was made was because the former template was not set up to handle solutions, and was missing various other pieces of data that were important in contemporary math. I made the decision with Phil to start treating injected exercises as a different feature from content exercises. This will impact any book that has injected exercises -- fortunately only 5 books for now! But hopefully it will help us be more flexible about injected exercises going forward, because they have a different set of features from content exercises and are apparently changed/updated often.

Reply:
Thank you for clarification! :relaxed: 
I tried again and it doesn't work for me :crying_cat_face: 

I run `./install-submodules.sh --upgrade`, then `cd recipes` and I checkout to `eng-comp-update`. After that I come back to `book-pipline` directory (`cd ..`) and I run

```
GH_SECRET_CREDS={token}:x-oauth-basic ./cli.sh ./data/english-composition/ all-git-pdf 'openstax/osbooks-writing-guide/writing-guide' english-composition main
```
This is what I get:

![image (5)](https://user-images.githubusercontent.com/40228854/132184202-8e578651-719a-4e0e-8860-17cb5e0b4a85.png)